src/Components/ExtractLevelsMenu.jsx

```diff
@@ -0,0 +1,186 @@
+        normal = new Vector3(0, 0, 1)
+        break
+      default:
+        normal = new Vector3(0, 1, 0)
+        break
+    }
+    console.log('in the function normal', normal)
+    if (!planeHash || planeHash !== normalDirection ) {
+      addHashParams(window.location, PLANE_PREFIX, {planeAxis: normalDirection})
+    }
+    return viewer.clipper.createFromNormalAndCoplanarPoint(normal, modelCenter)
+  }
+  useEffect(() => {
+    const planeHash = getHashParams(location, 'p')
+    if (planeHash && model && viewer) {
+      const modelCenter = getModelCenter(model)
+      const planeDirection = planeHash.split(':')[1]
+      let normal
+      switch (planeDirection) {
+        case 'x':
+          normal = new Vector3(1, 0, 0)
+          break
+        case 'y':
+          normal = new Vector3(0, 1, 0)
+          break
+        case 'z':
+          normal = new Vector3(0, 0, 1)
+          break
+        default:
+          normal = new Vector3(0, 1, 0)
+          break
+      }
+      console.log(' ------------------------- ')
+      console.log('the modelCenter from the useEffect', modelCenter)
+      console.log('plane direction', planeDirection)
+      console.log('normal', normal)
+      console.log(' ------------------------- ')
+      createPlane(planeDirection)
+      // viewer.clipper.createFromNormalAndCoplanarPoint(normal, modelCenter)
+    }
+  }, [model])
+
+  const showExtractMenu = async () => {
+    const allStor = await extractHeight(model)
+    console.log(allStor)
+
+    let sampleHeights = []
+
+
+    // PLACEHOLDER VALUES
+    /* eslint-disable */
+    const EISVOGEL = [
+      -2.24, 0, 3.5,
+      6.4, 9.5, 12.6,
+      15.7, 18.8, 21.81,
+      27.12,
+    ]
+  
+    /* eslint-enable */
+
+    // sampleHeights = EISVOGEL
+    sampleHeights = allStor
+    const sampleHeightsIndex = []
+    for (let i = 0; i < sampleHeights.length; i++) {
+      sampleHeightsIndex[i] = i
+    }
+    const floorplanMenu = []
+
+    const planeoffset = 0.5
+    sampleHeightsIndex.forEach((data) => {
+      floorplanMenu.push(
+          <MenuItem onClick={() => createFloorplanPlane(sampleHeights[data], sampleHeights[data + 1] - planeoffset)}>  L{data} </MenuItem>)
+    })
+    return floorplanMenu
+  }
+  showExtractMenu()
+
+
+  return (
+    <div>
+      <TooltipIconButton
+        title={'Extract Levels'}
+        icon={<LevelsIcon/>}
+        onClick={handleClick}
+      />
+      <Menu
+        elevation={1}
+        id='basic-menu'
+        anchorEl={anchorEl}
+        open={open}
+        onClose={handleClose}
+        anchorOrigin={{vertical: 'top', horizontal: 'center'}}
+        transformOrigin={{vertical: 'top', horizontal: 'center'}}
+        className={classes.root}
+        PaperProps={{
+          style: {
+            left: '300px',
+            transform: 'translateX(-40px) translateY(-40px)',
+          },
+        }}
+      >
+        <MenuItem onClick={() => planView()}> Plan View</MenuItem>
+        {showExtractMenu}
```

Comment:
@pablo-mayrgundter this is the part that I'm blocked with. `showExtractMenu` is an async operation as it awaits for `extractHeight` to calculate the levels. The goal is show the levels in the menu bar.